### PR TITLE
kde-threadweaver 5.71.0 (new formula)

### PR DESCRIPTION
**Pull request by yurikoles**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by alebcay** [comment]
```
==> brew test --verbose kde-threadweaver
==> FAILED
Testing kde-threadweaver
/usr/bin/sandbox-exec -f /private/tmp/homebrew20200523-78722-1q2oyee.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/kde-threadweaver.rb --verbose
==> cmake /usr/local/Cellar/kde-threadweaver/5.70.0/share/kde-threadweaver/examples/HelloWorld -Wno-dev
CMake Error at /usr/local/Cellar/cmake/3.17.2/share/cmake/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "FindQt5Core.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5Core", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5Core"
  (requested version 5.12.0) with any of the following names:

    Qt5CoreConfig.cmake
    qt5core-config.cmake

  Add the installation prefix of "Qt5Core" to CMAKE_PREFIX_PATH or set
  "Qt5Core_DIR" to a directory containing one of the above files.  If
  "Qt5Core" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /usr/local/lib/cmake/KF5ThreadWeaver/KF5ThreadWeaverConfig.cmake:28 (find_dependency)
  CMakeLists.txt:17 (find_package)
```

**Comment by yurikoles** [comment]
We have `depends_on "qt"`, it's not a build dependency, so why is it not being installed for testing bottle?

**Comment by yurikoles** [comment]
Hi @SMillerDev,

Sorry for ping. Could you please look into test issue? Why `qt` isn't found if it's a hard dependency and on my machine `brew test kde-threadweaver` even refuses to start if I force uninstall `qt`, but it passes successfully if I return it back.

**Comment by yurikoles** [comment]
Please add label `help needed`

**Comment by yurikoles** [comment]
@MikeMcQuaid sorry for ping, you are the only developer of `brew ci` I know.

Please review my first comment on `qt`

**Comment by yurikoles** [comment]
I saw similar failure in `kde-mac/kde` CI. If we solve it I will send a brunch of new formulas

File: Formula/kde-threadweaver.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    pkgshare.install "examples"
+  end
+
+  test do
+    system "cmake", (pkgshare/"examples/HelloWorld"), "-Wno-dev"
+    system "make"
+
+    expected_output = "Hello World!"
+    actual_output=shell_output("./ThreadWeaver_HelloWorld 2>&1").strip!
+    assert_equal expected_output, actual_output
```
Comment (MikeMcQuaid) [inline Formula/kde-threadweaver.rb:33]:
```suggestion
    assert_equal "Hello World!", shell_output("./ThreadWeaver_HelloWorld 2>&1").strip
```

**Comment by MikeMcQuaid** [comment]
> ==> brew fetch --retry bison cairo cmake doxygen fontconfig freetype fribidi gd gdbm gettext glib gobject-introspection graphite2 graphviz gts harfbuzz icu4c jasper jpeg kde-extra-cmake-modules libffi libpng libtiff libtool lzo meson netpbm ninja openssl@1.1 pango pcre pixman pkg-config python@3.8 qt readline sphinx-doc sqlite webp xz

Looks to me from this that Qt is being fetched and therefore probably installed. Can you try to add some debugging to the formula to verify it's definitely not being installed (rather than just not being found)?

**Comment by yurikoles** [comment]
@MikeMcQuaid as I see `brew test-bot` doesn't expose `qt` flags needed to build our test. Firstly I had tried to expose `Qt5_DIR`, but it seems that I need to advertise all required modules explicitly. I can obviously pick them from my verbose test log and modify them to be prefix-agnostic via `Formula["qt"]`, but this manual work looks frustrating, especially when the very same test just works locally.

**Comment by alebcay** [comment]
I wonder if it works locally because it's using something like `pkg-config` to find the necessary information for Qt5 linkage?

**Comment by yurikoles** [comment]
@alebcay it uses CMake, but for some reason it fails under `brew test`

**Comment by alebcay** [comment]
I was able to replicate the test failure locally with this formula (after commenting out the extra `cmake` flags in the test). So I'm a bit puzzled at why it's working for you.

I did notice this caveat on the `qt` formula:
```
qt is keg-only, which means it was not symlinked into /usr/local,
because Qt 5 has CMake issues when linked.
```

If I link the `qt` keg with `brew link --force qt` and rerun the test (having commented out the extra parts added to `args`), I get this error:
```
CMake Error at /usr/local/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:14 (message):
  The imported target "Qt5::Core" references the file

     "/usr/local/.//mkspecs/macx-clang"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/local/lib/cmake/Qt5Core/Qt5CoreConfigExtras.cmake"

  but not all the files it references.
```

Thus I think the best option is probably just to explicitly specify the CMake paths like you're doing already. I managed to reduce the number of flags we actually need to pass to CMake (I think - at least it worked locally, hopefully the CI will agree).

**Comment by yurikoles** [comment]
@alebcay 

Thanks for your valuable help! I had tried reinstalling `qt` many times during debugging of this formula test locally and I'm sure I didn't force-linked `qt`, it just works for me locally and doesn't work under CI :(. 

I will try to uninstall qt and run `brew prune`.

**Comment by alebcay** [comment]
I would also take a look in your shell environment (e.g. `.bashrc` or `.zshrc`) and see if there's anything related to Qt in there.

To be clear, I ran `brew test --verbose ./Formula/kde-threadweaver.rb` locally and produced the failure; I did not just run the commands in my regular shell environment.

**Comment by yurikoles** [comment]
> I would also take a look in your shell environment (e.g. `.bashrc` or `.zshrc`) and see if there's anything related to Qt in there.

Yes, you are right, I had 
```
export PATH="/usr/local/opt/qt/bin:${PATH}"
```

**Comment by yurikoles** [comment]
@alebcay could you please approve this formula?

**Comment by alebcay** [comment]
Now that `kde-extra-cmake-modules` 5.71.0 is merged, I will try bumping this up as well and if it passes, I think we'll be good to go. 🚀

**Review by alebcay — APPROVED** [review]

**Comment by alebcay** [comment]
Thanks for your contribution, @yurikoles! 🎊 Let me know if you would like any help with migrating other formulae from your tap at KDE Invent to here.

**Pull request closed without merging**